File: ros_learning_cpp/src/serviceServer.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;


class AddTwoInts_server: public rclcpp::Node{
    public:
    AddTwoInts_server():Node("Add_Two_Ints"){
        server_ = this->create_service<example_interfaces::srv::AddTwoInts>("the_addition_of_twoInts",
                    std::bind(&AddTwoInts_server::callback_addition_of_TwoInsts, this, _1, _2)); // service name, callbackfuc
        RCLCPP_INFO(this->get_logger(),"Add_Two_ints started...");

    }
    private:
    void callback_addition_of_TwoInsts(const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
                            const example_interfaces::srv::AddTwoInts::Response::SharedPtr response){
        response->sum = request->a +request->b;
        RCLCPP_INFO(this->get_logger(),"sum of %ld and %ld = %ld",request->a, request->b,response->sum);
            }
            
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;

};



int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoInts_server>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: ros_learning_pkg/ros_learning_pkg/serviceServer.py
#! /usr/bin/python3

import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts


class AddTwoInts_server(Node):
    def __init__(self):
        super().__init__("Add_Two_Ints")
        self.server = self.create_service(AddTwoInts, "the_addition_of_twoInts",
                                          self.callback_addition_of_TwoInts) # type of service, service name, callbackfnc,
        
        self.get_logger().info("Add_Two_Ints started...")
        
        
        
    def callback_addition_of_TwoInts(self, request, response):
        
        response.sum = request.a + request.b
        self.get_logger().info(f"sum of {request.a} and {request.b} = {response.sum}")
        return response
        
        
        
        
        
        
def main(args= None):
    rclpy.init(args= args)
    node = AddTwoInts_server()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()
    